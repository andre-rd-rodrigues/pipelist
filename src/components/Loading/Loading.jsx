import React from "react";
import { SyncLoader } from "react-spinners";
import styles from "./loading.module.scss";

const Loading = ({
  color = "rgb(54, 215, 183)",
  loading,
  size = 20,
  type = "SyncLoader",
  styling
}) => {
  const components = {
    SyncLoader
  };

  const Loader = components[type];

  return (
    loading && (
      <div style={styling} className={styles.loader} data-testid="loader">
        <Loader color={color} loading={loading} size={size} />
      </div>
    )
  );
};

export default Loading;
