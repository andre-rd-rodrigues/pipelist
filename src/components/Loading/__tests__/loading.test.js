import { render } from "@testing-library/react";
import reactTestRenderer from "react-test-renderer";
import Loading from "../Loading";

describe("Loader", () => {
  let props;

  props = {
    color: "rgb(54, 215, 183)",
    loading: false,
    size: 20,
    type: "SyncLoader"
  };

  it("should render correctly", () => {
    const wrapper = reactTestRenderer.create(
      <Loading {...props} loading={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should show loader when loading is true", () => {
    render(<Loading {...props} loading={true} />);

    expect(screen.getByTestId("sync_loader")).toBeInTheDocument();
  });

  it("should show proper spinner according to prop trype", () => {
    render(<Loading {...props} loading={true} />);

    expect(screen.getByTestId("sync_loader")).toBeInTheDocument();
  });
});
