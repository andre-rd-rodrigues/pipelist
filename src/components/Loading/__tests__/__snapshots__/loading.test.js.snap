// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<div
  className="loader"
  data-testid="loader"
>
  <span
    className="css-1xdhyk6"
  >
    <span
      className="css-1tbwygc"
    />
    <span
      className="css-x31nw7"
    />
    <span
      className="css-1en0n97"
    />
  </span>
</div>
`;
